src/TuDa.CIMS.Api/Repositories/PurchaseRepository.cs

```diff
@@ -0,0 +1,164 @@
+            .Include(i => i.Entries)
+            .ToListAsync();
+    }
+
+    /// <summary>
+    /// Returns a single purchase by its specific ID.
+    /// </summary>
+    /// <param name="id">the specific ID for the purchase</param>
+    /// <param name="workingGroupId">the specific id of a workinggroup</param>
+    /// <returns></returns>
+    public async Task<Purchase?> GetOneAsync(Guid id, Guid workingGroupId)
+    {
+        return await _context.WorkingGroups
+            .Where(i => i.Id == workingGroupId)
+            .SelectMany(i => i.Purchases)
+            .Where(i => i.Id == id)
+            .Include(i => i.Buyer)
+            .Include(i => i.Entries)
+            .SingleOrDefaultAsync();
+    }
+
+    /// <summary>
+    /// Updates an existing purchase by its ID using the provided update model.
+    /// </summary>
+    /// <param name="id">the specific ID of the purchase</param>
+    /// <param name="workingGroupId">the specific id of a workinggroup</param>
+    /// <param name="updateModel">the model containing the update values for the purchase </param>
+    /// <returns></returns>
+    public async Task<ErrorOr<Updated>> UpdateAsync(Guid id, Guid workingGroupId, UpdatePurchaseDto updateModel)
+    {
+        var existingItem = await _context
+            .WorkingGroups
+            .Where(i => i.Id == workingGroupId)
+            .SelectMany(i => i.Purchases)
+            .Where(i => i.Id == id)
+            .Include(i => i.Buyer)
+            .Include(i => i.Entries)
+            .SingleOrDefaultAsync();
+
+        if (existingItem is null)
+        {
+            return Error.NotFound("Purchase.update", $"Purchase with ID {id} was not found.");
+        }
+
+
+        if (updateModel.Buyer is not null)
+        {
+            var buyer = await _context.Persons.SingleOrDefaultAsync(r => r.Id == updateModel.Buyer.Id);
+            if (buyer is null)
+            {
+                return Error.NotFound("Purchase.update", $"Person with ID {updateModel.Buyer.Id} was not found.");
+            }
+
+            existingItem.Buyer = buyer;
+        }
+
+        if (updateModel.Entries is not null)
+        {
+            var entries = await _context.PurchaseEntries
+                .Where(x => updateModel.Entries.Select(y => y.Id).Contains(x.Id)).ToListAsync();
+            if (entries.Count != updateModel.Entries.Count)
+            {
+                return Error.NotFound("Purchase.update", $"Some of the PurchaseEntries were not found.");
+            }
+
+            existingItem.Entries = entries;
+        }
+
+        existingItem.Buyer = updateModel.Buyer ?? existingItem.Buyer;
+        existingItem.Signature = updateModel.Signature ?? existingItem.Signature;
+        existingItem.Entries = updateModel.Entries ?? existingItem.Entries;
+        existingItem.CompletionDate = updateModel.CompletionDate ?? existingItem.CompletionDate;
+        existingItem.Completed = updateModel.Completed ?? existingItem.Completed;
```

Comment:
🔧 🌱  Buyer and Entries should be removed, and `CompletionDate` and `Complete` should only be updatable together. Good Practice would be creating a Complete endpoint.